app/models/solidus_volume_pricing/pricer.rb

```diff
@@ -7,18 +7,22 @@ def self.pricing_options_class
     end
 
     def price_for(pricing_options)
-      @quantity = pricing_options.quantity
-      @user = pricing_options.user
+      extract_options(pricing_options)
+      Spree::Money.new(computed_price)
+    end
 
-      if volume_prices.empty?
-        variant.display_price
-      else
-        Spree::Money.new(computed_price)
-      end
+    def earning_amount(pricing_options)
+      extract_options(pricing_options)
+      Spree::Money.new(computed_earning)
     end
 
     private
 
+    def extract_options(pricing_options)
+      @quantity = pricing_options.quantity
```

Comment:
This can lead to questionable behaviour if people aren't paying attention. Usually, we only set instance variables in initialization. 

Doing it elsewhere without the method name explicitly stating that it is "setting" something on the class, or for memoization (usually we indicate memoization with @_somthing to indicate we're not using the instance variable for anything else)

Moreso the method name indicates that it should return "options", either as an array, a hash, or a structure.

It's okay to not set this as instance variables, and make private methods take an argument.